readme: updates (#2120)
Unvanquished/Unvanquished: README.md

If cmake complains about missing files in `daemon/` folder or similar issue then you have skipped this step.

## Downloading the game's assets

If you don't have the assets, you can download them first. You can download an universal zip from the [download page](https://unvanquished.net/download/), here is the [link to the latest version](https://unvanquished.net/download/zip). The `pkg/` folder would have to be stored next to your `daemon` binary.

> Otherwise you can use the package downloader script. This script can use `aria2c`, `curl` or `wget`, `aria2c` is recommended.

Reviewer: ```suggestion
Otherwise you can use the package downloader script. This script can use `aria2c`, `curl` or `wget`. `aria2c` is recommended.
```

Author: Done

```

You have to put all your custom files in a dpkdir. For example we will name the mod “_mod-custom_” so the dpkdir will be stored as `pkg/mod-custom_src.dpkdir`.

```sh
cd build
mkdir -p pkg/mod-custom_src.dpkdir
```

Copy to this directory all your custom files to be shipped with the mod.

If you built custom game logic binaries, copy the `.nexe` files in that folder.

```sh
cp [cs]game-*[46].nexe pkg/mod-custom_src.dpkdir
```

Set Unvanquished as a dependency of your mod by writing a `DEPS` file in the dpkdir containing the `unvanquished` string. You have to do that if your mod contains custom game logic binaries but relies on files from `unvanquished_src.dpkdir` you don't distribute yourself, or your mod contains other custom files but you don't distribute custom game logic binaries.

```sh
echo 'unvanquished' > pkg/mod-custom_src.dpkdir/DEPS
```

Assuming the Unvanquished released assets are stored in a `pkg/` folder next to the `daemon` binary, you can then test your mod this way (loading it as `nexe` binary):

```sh
./daemon set fs_extrapaks mod-custom +map plat23
```

You can use `-pakpath` to load base assets from a `pkg/` folder stored somewhere else (see above).

> For debugging purpose, you can run your mod as dll:

Reviewer: ```suggestion
For debugging purpose, you can run your mod as a shared library:
```

Author: Done

```sh
./daemon set fs_extrapaks mod-custom \
			-set vm.sgame.type 3 -set vm.cgame.type 3 \
			+map plat23
```

Once you've successfully tested your code as `nexe` binary, you create a dpk archive, this is a classic zip archive with a `.dpk` file extension. The zip should not contain the dpkdir directory but the content of it:

```sh
( cd pkg/mod-custom_src.dpkdir ; zip ../mod-custom_0.1.dpk . )
```

The dpk file is named this way: `<BASENAME>_<VERSION>.dpk`. Here the basename is `mod-custom` and the version is `0.1`. The version string for dpk packages better start with a number, see [Packaging version guidelines](https://wiki.unvanquished.net/wiki/Packaging_version_guidelines) on the wiki for details. See also the [dpk format page ](https://wiki.unvanquished.net/wiki/Formats/DPK) on the wiki for details about this format.

You then start a game server with your custom mod this way:

```sh

> ./daemonded set fs_extrapaks mod-custom +map plat23

Reviewer: ```suggestion
./daemonded -set fs_extrapaks mod-custom +map plat23
```
(or maybe `+set` works too)

Author: Done
